kotlinx-coroutines-core/native/src/MultithreadedDispatchers.kt

```diff
@@ -63,28 +64,54 @@ internal class WorkerDispatcher(name: String) : CloseableCoroutineDispatcher(),
     }
 }
 
-private class MultiWorkerDispatcher(name: String, workersCount: Int) : CloseableCoroutineDispatcher() {
+private class MultiWorkerDispatcher(
+    private val name: String,
+    private val workersCount: Int
+) : CloseableCoroutineDispatcher() {
     private val tasksQueue = Channel<Runnable>(Channel.UNLIMITED)
-    private val workers = Array(workersCount) { Worker.start(name = "$name-$it") }
-
-    init {
-        workers.forEach { w -> w.executeAfter(0L) { workerRunLoop() } }
-    }
+    private val workers = atomicArrayOfNulls<Worker>(workersCount)
+    private val activeWorkers = atomic(0)
 
     private fun workerRunLoop() = runBlocking {
         for (task in tasksQueue) {
-            // TODO error handling
+            /**
+             * Any unhandled exception here will pass through worker's boundary and will be properly reported.
+             */
             task.run()
         }
     }
 
     override fun dispatch(context: CoroutineContext, block: Runnable) {
-        // TODO handle rejections
-        tasksQueue.trySend(block)
+        if (activeWorkers.value != workersCount) {
+            tryAddWorker()
```

Comment:
I would really like to provide a proper keep-alive here, but it doesn't seem adequately possible: we **must** terminate workers and we **must** await termination in a blocking way, otherwise native resources leak.

The preliminary decision is to get rid of workers and design something better (i.e. `Thread`) in K/N